src/webgl/3d_primitives.js

```diff
@@ -1588,13 +1588,62 @@ p5.RendererGL.prototype.bezierVertex = function(...args) {
 
     const LUTLength = this._lookUpTableBezier.length;
 
+    // Get the last fill color
+    const _vertexColors = this.immediateMode.geometry.vertexColors;
+    const _vertexColorSize = _vertexColors.length;
+    const lastFillColor = [
+      _vertexColors[_vertexColorSize - 4],
+      _vertexColors[_vertexColorSize - 3],
+      _vertexColors[_vertexColorSize - 2],
+      _vertexColors[_vertexColorSize - 1]
+    ];
+    const finalFillColor = this.curFillColor.slice();
+
     if (argLength === 6) {
       this.isBezier = true;
 
       w_x = [this.immediateMode._bezierVertex[0], args[0], args[2], args[4]];
       w_y = [this.immediateMode._bezierVertex[1], args[1], args[3], args[5]];
+      // calCalculate intermediate colors
```

Comment:
Maybe we can also describe in this comment, at a high level, the strategy we're using to pick intermediate colours? Something about using the fraction of the distance along the control points

Reply:
yeah that sounds good!